app/lib/shared/platform.dart

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:pana/pana.dart';
+
+abstract class KnownPlatforms {
+  static const String flutter = PlatformNames.flutter;
+  static const String server = PlatformNames.server;
+  static const String web = PlatformNames.web;
+  static const List<String> everywhere = const [flutter, server, web];
+
+  static bool isPredicate(String platform) => everywhere.contains(platform);
+}
+
+List<String> indexDartPlatform(DartPlatform platform) {
+  if (platform == null) {
+    return null;
+  }
+  if (platform.worksEverywhere) {
+    return KnownPlatforms.everywhere;
+  }
+  if (platform.restrictedTo == null || platform.restrictedTo.isEmpty) {
+    return null;
+  }
+  return new List.from(platform.restrictedTo);
```

Comment:
Side note: `DartPlatform` seems also like not the best name, since it's not really an identifier but behaves more like a set/list.

Reply:
Yeah, I agree it is a crappy name, and I'll be happy to refactor it if there is a better one. Platform classification has way too many edge cases, so it either gets too simple (leaving out legitimate concerns), or somewhat confusing (like the current state). I've erred on the later, because I think the "which platform will this be running on" will be an important info (icons, search filters), and I'd rather have a bit more complexity in how we handle it, but get it correct from the user's point of view.